src/llama.cpp

```diff
@@ -5640,7 +5640,7 @@ static void llm_load_vocab(
     // build special tokens cache
     {
         for (llama_vocab::id id = 0; id < (llama_vocab::id)n_vocab; ++id) {
-            if (!(vocab.id_to_token[id].attr & LLAMA_TOKEN_ATTR_NORMAL)) {
+            if (vocab.id_to_token[id].attr & (LLAMA_TOKEN_ATTR_CONTROL | LLAMA_TOKEN_ATTR_USER_DEFINED)) {
```

Comment:
Should UNKNOWN tokens only be parsed when `parse_special == true`, or all the time like `USER_DEFINED` tokens?

I'm not sure if UNKNOWN tokens should be specially parsed at all.

I would tend toward only parsing them with `parse_special == true`, like CONTROL tokens. (this is also the current behavior of `master`)

I'm trying to figure out where UNKNOWN tokens are used and if it's useful to specially parse them. But this might differ from HF's `tokenizers`, so I need a model with UNKNOWN tokens to test this out. If I don't find one, I can always modify one.

Reply:
> but we need parse UNKNOWN token.

I've fixed this in <https://github.com/ggerganov/llama.cpp/commit/98edea60bcd67458d70138c13c428414cd8ec63a>. UNKNOWN tokens are parsed when `parse_special == true`, as on `master`.